iceoryx_hoofs/test/moduletests/test_concurrent_lockfree_queue.cpp

```diff
@@ -39,6 +40,14 @@ struct Integer
     {
     }
 
+    // Delete copy constructor to ensure that the queue supports move-only types.
+    Integer(const Integer&) = delete;
+    Integer& operator=(const Integer&) = delete;
+    Integer(Integer&&) = default;
+    Integer& operator=(Integer&&) = default;
+
+    ~Integer() = default;
+
```

Comment:
Could you revert this and create a `IntegerMoveOnly` and `IntegerCopyOnly` struct (maybe inherit from `integer`) and add two tests for each of the two structs for `push` and `tryPush`, e.g.
```cpp
TEST(LockFreeQueueMoveTest, PushPopWithMoveOnlyTypeWorks)
{
    ::testing::Test::RecordProperty("TEST_ID", "dc85a403-8265-4853-8360-73ea66668f24");

    iox::concurrent::LockFreeQueue<IntegerMoveOnly, 10> sut;
    const int EXPECTED_VALUE{42};
    sut.push({EXPECTED_VALUE});
    sut.pop().and_then([&](const auto& value) { EXPECT_THAT(value, Eq(EXPECTED_VALUE)); }).or_else([&] {
        GTEST_FAIL() << "Expected '" << EXPECTED_VALUE << "' but got nothing!";
    });
}
```
Could be done before `LockFreeQueueTest` class definition.

The other tests can remain as they are since they test the internal logic and if `push`, `try_push` and `pop` work with move only types everything will work.

Reply:
I certainly can and will refactor to add additional distinct `MoveOnly` and `CopyOnly` versions. 

I wonder however if it would not be more beneficial to simply add those types to the existing type list so they get included in all of the existing tests. Is the idea here to preserve the integrity of the existing tests?

Reply:
You can of course add them to the type list.  At a first glance the setup looked a bit complicated and I was not sure how easy it would be to add tests with a different element type. It's up to you :)

Reply:
It turned out to be pretty easy. Just define the new type and add it to a bunch of templates. I confirmed by undoing the `std::forward` fix briefly to observe the failure recurring.